### PR TITLE
graalvm-oracle: init at 22

### LINKED ISSUE #259631: Package request: Oracle Graalvm 21
**Project description**

Oracle Graalvm is different than Graalvm Community Edition, it has more options for graalvm tuning performance like `--pgo`.

**Metadata**

* homepage URL: https://www.oracle.com/java/graalvm/
* source URL: https://download.oracle.com/graalvm/21/archive/graalvm-jdk-21_linux-x64_bin.tar.gz
* license: custom oracle license?
* platforms: linux

### PR DESCRIPTION
**Pull request by farnoy**
Supersedes #259639, resolves #259631

## Description of changes

This adds oracle-graalvm and make buildGraalvm more generic to be used on other places than community-edition.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Pinging @thiagokokada (reviewer from the previous PR)

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (farnoy) [inline pkgs/top-level/all-packages.nix:16086]: This used to be called GraalVM EE, now just "Oracle GraalVM". Maybe `oracle-graalvm` would be a better name here?

**Comment by rafaelrc7** [comment]
Do you plan on adding the other supported versions by oracle's graalvm? Such as for JDK 17 and 21.

**Comment by farnoy** [comment]
@rafaelrc7 I don't have any use for them but I'll take a look

**Comment by farnoy** [comment]
Rebased on latest nixpkgs to resolve conflicts

**Comment by rafaelrc7** [comment]
> @rafaelrc7 I don't have any use for them but I'll take a look

Thanks! I am currently working on packaging pkl in #286658 and the native build depends on graalvm 17, so it would be a lot of help.

File: pkgs/top-level/all-packages.nix
Comment (thiagokokada) [inline pkgs/top-level/all-packages.nix:15883]: I don't like that we have 2 different `graalvmPackages`.

I think since we are introducing Oracle GraamVM again, it is a good time to move everything to `graalvmPackages` and make `graalvmCEPackages` an alias to `graalvmPackages`, and deprecate and eventually remove `graalvmCEPackages`.

Comment (farnoy) [reply pkgs/top-level/all-packages.nix:15883]: I can do that, but can you be more specific? Do we want `graalvmPackages` to temporarily be the CE packages and also `graalvmPackages.oracle` + `graalvmPackages.ce` as the long-term, stable names?

Do we also want this nesting for JDK packages or is `pkgs.oracle-graalvm` fine as it is in the current version of the PR?

Comment (thiagokokada) [reply pkgs/top-level/all-packages.nix:15883]: > I can do that, but can you be more specific? Do we want graalvmPackages to temporarily be the CE packages and also graalvmPackages.oracle + graalvmPackages.ce as the long-term, stable names?

Yes. But probably the names will be `graalvmPackages.graalvm-oracle` and `graalvmPackages.graalvm-ce`, because there are other packages inside `graalvmCEPackages` too.

Comment (farnoy) [reply pkgs/top-level/all-packages.nix]: I've updated the PR, please let me know what you think

File: pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix
Comment (farnoy) [inline pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:26]: Not sure if this URL is deterministic, there is no `/archive/` version for these

File: pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix
Comment (farnoy) [inline pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:9]: I think these hashes are correct, I calculated them with nix-prefetch-url but I need to double check.

File: pkgs/by-name/gr/graalvm-ce/package.nix
Comment (farnoy) [inline pkgs/by-name/gr/graalvm-ce/package.nix:2]: These are meant to be simple aliases, temporary at that, as discussed. I did it this way so that all-packages.nix doesn't need to be let recursive as that would be a huge change potentially.

File: pkgs/by-name/gr/graalvmCEPackages/package.nix
Comment (chayleaf) [inline pkgs/by-name/gr/graalvmCEPackages/package.nix:1]: this file should be removed

File: pkgs/by-name/gr/graalvm-ce/package.nix
Comment (chayleaf) [inline pkgs/by-name/gr/graalvm-ce/package.nix:1]: can't you use aliases.nix?

File: pkgs/by-name/gr/graalvm-ce/package.nix
Comment (chayleaf) [reply pkgs/by-name/gr/graalvm-ce/package.nix]: isn't it already? (basically) all-packages.nix uses `with pkgs;` at the top

File: pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix
Comment (chayleaf) [inline pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:3]: why not flatten `hashes` attr into the parent attrset? also, why not use SRI hashes (`hash` instead of `sha256`)?

**Comment by farnoy** [comment]
Thanks for the feedback @chayleaf , updated

File: pkgs/development/compilers/graalvm/default.nix
Comment (chayleaf) [inline pkgs/development/compilers/graalvm/default.nix:18]: something like

```suggestion
  graalvm-oracle-22 = callPackage ./graalvm-oracle { version = "22"; };
  graalvm-oracle = graalvm-oracle-22;
```

File: pkgs/development/compilers/graalvm/default.nix
Comment (chayleaf) [inline pkgs/development/compilers/graalvm/default.nix:5]: github actions failed here

```suggestion
  callPackage,
```

File: pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix
Comment (chayleaf) [inline pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:4]:
```suggestion
      hash = "sha256-0wPTDBB2T+qrB27+eQGd88mBsEh/ysPfelMT929hBA4=";
```

(and the same below)

File: pkgs/top-level/aliases.nix
Comment (chayleaf) [inline pkgs/top-level/aliases.nix:566]:
```suggestion
  graalvmCEPackages = graalvmPackages; # 2024-08-10
```

aliases.nix is for deprecated aliases, so you have to put the date in there as a rough estimate of whether it's okay to replace the alias with a `throw` (see ./maintainers/scripts/remove-old-aliases.py)

graalvm-ce is probably fine to not deprecate? in that case, just put it back to all-packages.nix

File: pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix
Comment (thiagokokada) [inline pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:5]: Those URLe are unacceptable, especially in this case since Hydra will not cache the result (because this is unfree).

We need either to have a stable URL from upstream or have the files archived somewhere (e.g.: archive.org).

Comment (chayleaf) [reply pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:5]: there are some precedents, but they typically result in the packages being marked as broken (see fmod)

Comment (farnoy) [reply pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:5]: I found more stable links (with full version strings) [here](https://www.oracle.com/java/technologies/javase/graalvm-jdk22-archive-downloads.html).

I'm not sure if this is sufficient from what you're saying. This package will always be unfree, regardless where we download it from? And I'm not sure it's legal to replicate it to archive.org.

Comment (chayleaf) [reply pkgs/development/compilers/graalvm/graalvm-oracle/hashes.nix:5]: the ones with full version strings should be sufficient

File: pkgs/top-level/aliases.nix
Comment (farnoy) [reply pkgs/top-level/aliases.nix]: `graalvm-ce` now lives in `graalvmPackages.graalvm-ce` so  I'm deprecating this old name https://github.com/NixOS/nixpkgs/pull/321026#discussion_r1692035568

**Comment by farnoy** [comment]
Updated download links to those for a specific patch version of each package.

File: pkgs/top-level/all-packages.nix
Comment (thiagokokada) [inline pkgs/top-level/all-packages.nix:15882]: I don't like the usage of `recurseIntoAttrs` here, we probably need to introduce a scope inside `graalvmPackages` to ensure that all packages are build with the same scope.

Comment (farnoy) [reply pkgs/top-level/all-packages.nix:15882]: I'm not familiar with scopes and not sure what this means. I tried just removing `recurseIntoAttrs` and things seem to work all the same, so it seems it can be safely dropped?

Is the fact that `pkgs/development/compilers/graalvm/graalvm-oracle/default.nix` uses `graalvmPackages.buildGraalvm` (from the parent "scope"?) problematic?

**Comment by farnoy** [comment]
Anything I should do to push this forward?

**Comment by thiagokokada** [comment]
@ofborg build babashka clojure-lsp

**Comment by thiagokokada** [comment]
Build broke for `graalvm` in darwin. Need to investigate if this is something that is currently broken in `master` or is something that this branch is doing.

**Comment by farnoy** [comment]
I've had a friend build clojure-lsp and babashka off this branch on aarch64-darwin, and it worked. Not sure what's going wrong when ofborg does it?

```
nix run github:nixos/nixpkgs#nix-info -- -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Review by thiagokokada — CHANGES_REQUESTED** [review]
Please squash your commits accordingly to [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) guidelines.

**Comment by farnoy** [comment]
@thiagokokada Squashed and updated the commit author to myself

**Comment by farnoy** [comment]
Resolved conflicts again

**Comment by farnoy** [comment]
Friendly ping :)

**Comment by farnoy** [comment]
I think this is a good test but I don't think I can use ofborg

@ofborg build graalvm-oracle graalvmPackages.graalvm-oracle_17 graalvm-ce graalvmPackages.graaljs cq

**Comment by farnoy** [comment]
nvm, it doesn't evaluate unfree packages

File: pkgs/development/compilers/graalvm/default.nix
Comment (thiagokokada) [inline pkgs/development/compilers/graalvm/default.nix:13]: I think the community edition should be the default here and the Oracle should be the other.

But actually maybe this separation doesn't make sense at all. We can just put all packages in one file I think.

File: pkgs/development/compilers/graalvm/default.nix
Comment (thiagokokada) [inline pkgs/development/compilers/graalvm/default.nix:11]: You are adding 2 separate versions but the commit only says `graalvm-oracle: init at 22`.

File: pkgs/development/compilers/graalvm/default.nix
Comment (farnoy) [reply pkgs/development/compilers/graalvm/default.nix:13]: There isn't a default, they're both in `graalvmPackages`, with extra aliases outside. It's `graalvm-ce` (same as before) and `graalvm-oracle`.

I will merge  `community-edition/default.nix` into this one, and we can have just one scope for everything

File: pkgs/development/compilers/graalvm/default.nix
Comment (farnoy) [reply pkgs/development/compilers/graalvm/default.nix]: I've split it up into two commits. I'll do some updates and introduce version 23 in a follow up PR

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 321026`

---
### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>babashka</li>
    <li>babashka-unwrapped</li>
    <li>bbin</li>
    <li>neil</li>
    <li>tests.writers.bin.babashka</li>
    <li>tests.writers.simple.babashka</li>
    <li>tests.writers.wrapping.babashka</li>
    <li>tests.writers.wrapping.babashka-bin</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>certificate-ripper</li>
    <li>clj-kondo</li>
    <li>cljfmt</li>
    <li>clojure-lsp</li>
    <li>cq</li>
    <li>dapl-native</li>
    <li>dbqn-native</li>
    <li>graalvmPackages.graaljs</li>
    <li>graalvmPackages.graalnodejs</li>
    <li>graalvmPackages.graalpy</li>
    <li>graalvmPackages.graalvm-ce</li>
    <li>graalvmPackages.graalvm-ce-musl</li>
    <li>graalvmPackages.graalvm-oracle (graalvmPackages.graalvm-oracle_22)</li>
    <li>graalvmPackages.graalvm-oracle_17</li>
    <li>graalvmPackages.truffleruby</li>
    <li>jet</li>
    <li>scala-update</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
    <li>yamlscript</li>
    <li>zprint</li>
  </ul>
</details>

**Comment by farnoy** [comment]
@thiagokokada 

So babashka fails on this test: https://github.com/NixOS/nixpkgs/blob/394571358ce82dff7411395829aa6a3aad45b907/pkgs/development/interpreters/babashka/default.nix#L38C1-L39C1

```$out/bin/bb '(prn "bépo àê")' | fgrep 'bépo àê'```

It outputs `"b?po ??"` on my end, is my system's locale/encoding leaking somewhere? 

I tried it on my other NixOS machine and it works:

```
~$ nix build "github:farnoy/nixpkgs?branch=oracle-graalvm#babashka"
~ 35s $ nix eval "github:farnoy/nixpkgs?branch=oracle-graalvm#babashka"
«derivation /nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv»
~$ nix log /nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv | grep '"b'
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv^*'
"bépo àê"
```

I don't know how to debug this deeper than this, but I think something from the environment is leaking

File: pkgs/development/compilers/graalvm/buildGraalvm.nix
Comment (thiagokokada) [inline pkgs/development/compilers/graalvm/buildGraalvm.nix:1]: This file is out of date with the changes from `master`.

Please update it, once it has those changes it should work: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix#L148-L172.

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 321026`

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>dapl-native</li>
    <li>dbqn-native</li>
    <li>obb</li>
  </ul>
</details>
<details>
  <summary>:x: 17 packages failed to build:</summary>
  <ul>
    <li>babashka</li>
    <li>babashka-unwrapped</li>
    <li>bbin</li>
    <li>certificate-ripper</li>
    <li>clj-kondo</li>
    <li>cljfmt</li>
    <li>clojure-lsp</li>
    <li>cq</li>
    <li>graalvmPackages.graalvm-ce</li>
    <li>graalvmPackages.graalvm-oracle</li>
    <li>graalvmPackages.graalvm-oracle_17</li>
    <li>jet</li>
    <li>neil</li>
    <li>scala-update</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
    <li>yamlscript</li>
    <li>zprint</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>graalvmPackages.graaljs</li>
    <li>graalvmPackages.graalnodejs</li>
    <li>graalvmPackages.graalpy</li>
    <li>graalvmPackages.truffleruby</li>
  </ul>
</details>

File: pkgs/development/compilers/graalvm/buildGraalvm.nix
Comment (farnoy) [reply pkgs/development/compilers/graalvm/buildGraalvm.nix:1]: This is the parent of my commits https://github.com/NixOS/nixpkgs/commit/3b74ae165f2bf31bcfd2a6c7c101ad50392e165d

I don't think I'm missing anything

**Comment by farnoy** [comment]
@ofborg build jet babashka clj-kondo

I don't have a darwin machine to test locally

**Comment by farnoy** [comment]
There's something non-deterministic about it. I'm able to build babashka on my branch, on the machine that got the encoding error before.

```
$ nix build "github:farnoy/nixpkgs?branch=oracle-graalvm#babashka" --no-substitute
1m 22.9s $ nix eval "github:farnoy/nixpkgs?branch=oracle-graalvm#babashka"«derivation /nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv»
$ nix log /nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/18ikvgm7qy41vkwhjyc8zmvyhic2i28a-babashka-1.12.194.drv^*'
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/rc83vlda57811c6yld7hb6f70b6z9d9s-babashka-1.12.194
checking for references to /build/ in /nix/store/rc83vlda57811c6yld7hb6f70b6z9d9s-babashka-1.12.194...
patching script interpreter paths in /nix/store/rc83vlda57811c6yld7hb6f70b6z9d9s-babashka-1.12.194
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
babashka v1.12.194
3
[1 2]
"bépo àê"
"ä\n"
#object["[Ljava.lang.Object;" 0x2c040989 "[Ljava.lang.Object;@2c040989"]
Clojure CLI version (deps.clj) 1.12.0.1479
```

File: pkgs/development/compilers/graalvm/buildGraalvm.nix
Comment (thiagokokada) [reply pkgs/development/compilers/graalvm/buildGraalvm.nix:1]: Just compare this file with the one from `master`: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/graalvm/community-edition/buildGraalvm.nix.

It is pretty clear that something went wrong. Probably because you moved the file before the changes.

Comment (thiagokokada) [reply pkgs/development/compilers/graalvm/buildGraalvm.nix:1]: Oh I know the reason, your PR introduces another file called `buildGraalvm` in another path and forgot to delete the previous one (the one inside `community-edition`). So it didn't get any conflicts and now we are building from the wrong file.

**Comment by thiagokokada** [comment]
> There's something non-deterministic about it. I'm able to build babashka on my branch, on the machine that got the encoding error before.

This non-deterministic behavior was fixed a long time ago in `master` in `buildGraalvm` but your PR doesn't reflect those changes. Please update it to use the correct version of the file.

File: pkgs/development/compilers/graalvm/buildGraalvm.nix
Comment (thiagokokada) [reply pkgs/development/compilers/graalvm/buildGraalvm.nix:1]: Or in other words, this should appear in the diff as a move, not as a new file.

**Comment by farnoy** [comment]
Good catch, thanks. That file is gone now and there's only the one `buildGraalvm.nix` that's fully up to date with master

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 321026`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>graalvmPackages.graaljs</li>
    <li>graalvmPackages.graalnodejs</li>
    <li>graalvmPackages.graalpy</li>
    <li>graalvmPackages.graalvm-ce</li>
    <li>graalvmPackages.graalvm-ce-musl</li>
    <li>graalvmPackages.graalvm-oracle</li>
    <li>graalvmPackages.graalvm-oracle_17</li>
    <li>graalvmPackages.truffleruby</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]
Code-wise, LGTM.

Will run `nixpkgs-review` for `aarch64-darwin` and `x86_64-darwin`, and also wait until the eval checks pass to merge.

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 321026`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>graalvmPackages.graaljs</li>
    <li>graalvmPackages.graalnodejs</li>
    <li>graalvmPackages.graalpy</li>
    <li>graalvmPackages.graalvm-ce</li>
    <li>graalvmPackages.graalvm-oracle</li>
    <li>graalvmPackages.graalvm-oracle_17</li>
    <li>graalvmPackages.truffleruby</li>
  </ul>
</details>

**Comment by farnoy** [comment]
What a journey! Thanks. I don't know what's going  with ofborg, I haven't seen it be green anytime recently

**Comment by thiagokokada** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 321026`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>graalvmPackages.graaljs</li>
    <li>graalvmPackages.graalnodejs</li>
    <li>graalvmPackages.graalpy</li>
    <li>graalvmPackages.graalvm-ce</li>
    <li>graalvmPackages.graalvm-oracle</li>
    <li>graalvmPackages.graalvm-oracle_17</li>
    <li>graalvmPackages.truffleruby</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
> What a journey! Thanks. I don't know what's going  with ofborg, I haven't seen it be green anytime recently

It is slower, but still running. It should take 1 or 2 days to eval your PR, but it should happen eventually.

**Comment by farnoy** [comment]
@thiagokokada it's green, can you merge?

**Pull request merged**
